### ISSUE TITLE
`importlib.metadata` issue on import (Python 3.7 & Linux)

### ISSUE THREAD

**Issue by uschmidt83** [napari/napari#4747]
It looks like importing napari 0.4.14 with Python 3.7 on Ubuntu 20.04.4 causes a `ValueError` [here](https://github.com/napari/napari/blob/v0.4.14/napari/utils/misc.py#L65).
See [this GitHub action log](https://github.com/stardist/stardist-napari/runs/7068816275?check_suite_focus=true#step:3:600) for all the details and how to reproduce.

Note: This issue does not occur on Windows or macOS.

**Comment by cnstt** [comment]
I don't know if this issue might be worth trying to fix, because napari 0.4.14 is an old version (current version is 0.4.16) and Python 3.7 support has been dropped since napari release 0.4.15 https://github.com/napari/napari/releases/tag/v0.4.15

**Comment by uschmidt83** [comment]
I respectfully disagree. If napari claims to support Python 3.7 in version 0.4.14, then it should work.
Also, version 0.4.14 came out on Feb 04, 2022, which is not that old...

**Comment by cnstt** [comment]
> I respectfully disagree. If napari claims to support Python 3.7 in version 0.4.14, then it should work.
> Also, version 0.4.14 came out on Feb 04, 2022, which is not that old...

I agree that it should work, but maybe we could figure out what is blocking you to update to a newer release? I don't know the maintenance policy at napari (I'm not a core developer :laughing:), but I don't know if they continue maintaining past versions... I mean even though it's not that old, I'm imagining myself the burden It would be to maintain several versions (including old ones) at the same time. Maybe ping @alisterburt for more info ;)

**Comment by uschmidt83** [comment]
> I agree that it should work, but maybe we could figure out what is blocking you to update to a newer release?

This is not about me personally. I maintain a plugin that's (currently) Python 3.7 compatible and noticed this issue when my daily tests against the latest version of napari (for the respective version of Python) failed.

**Comment by cnstt** [comment]
Ah ok I see :) So maybe to raise the minimum required python version to 3.8 would solve the issue in your plugin here?

**Comment by alisterburt** [comment]
Thanks for the report @uschmidt83 and for your input @cnstt !

As it says in the linked release notes, we aim to follow [NEP29](https://numpy.org/neps/nep-0029-deprecation_policy.html) in line with the big packages in the scientific python stack to reduce maintenance burden. 

I'm not sure why the bug has come up now... we do not yet have the capacity to support older napari versions actively and as we don't pin dependencies unless necessary older versions are likely to break. If you aren't experiencing problems with the current napari release on supported Python versions I don't see a reason to spend time fixing here so will close, please feel free to reopen if you would like to discuss further! 🙂

**Comment by Czaki** [comment]
Hello @uschmidt83. I also observe this problem in PartSeg but do not have time ealaier to debug this. 

The regression is in `importlib_metadata`, not in `napari` itself. In version `importlib-metadata==4.12.0` that was released on the 25 of Jun, they change the behavior for the empty string as input. 

The fastest solution for you will be to block this version of `importlib_metadata` for your CI. 

I have opened https://github.com/python/importlib_metadata/issues/395 for this. 

Because of https://github.com/python/cpython/issues/93259 I reopen this issue, as https://github.com/napari/napari/blob/23c85bfd76a0eb40622eb2dfeb4efdc19cf1c58d/napari/utils/misc.py#L67 is still in napari codebase I reopen this issue, because we need to add a check if `app_module` is `None` and support it independently.
